src/main/java/com/socialmedia/service/UserService.java

```diff
@@ -223,4 +224,11 @@ public FriendshipStatus checkFriendshipStatus(String targetUsername) {
     }
     return FriendshipStatus.NOT_FRIENDS;
   }
+
+  public Page<ApplicationUser> getActiveFriends(Pageable pageable) {
+    Principal principal = SecurityContextHolder.getContext().getAuthentication();
+    ApplicationUser user = getById(principal.getName());
+    long activeTime = System.currentTimeMillis() - 300000;
```

Comment:
let save it to some constant smth like ONLINE_LAST_ACTIVITY_THRESHOLD or any other variable name you like